models/errors.go

```diff
@@ -0,0 +1,88 @@
+package models
+
+import (
+	"context"
+	"errors"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+)
+
+type ErrorResponse struct {
+	Errors []BaseError `json:"errors"`
+	Status int         `json:"-"`
+}
+
+type BaseError interface {
```

Comment:
II think error is already an interface in go, so I'm not actually sure this interface type adds value. On L32, you could just return `error` as the type and other than it being a pointer I think your custom types would be valid